src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -42,52 +45,75 @@
   : Qt3DRender::QGeometry( parent )
   , mPositionAttribute( new Qt3DRender::QAttribute( this ) )
-  , mClassAttribute( new Qt3DRender::QAttribute( this ) )
+  , mParameterAttribute( new Qt3DRender::QAttribute( this ) )
+  , mColorAttribute( new Qt3DRender::QAttribute( this ) )
 #if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
   , mVertexBuffer( new Qt3DRender::QBuffer( Qt3DRender::QBuffer::VertexBuffer, this ) )
 #else
   , mVertexBuffer( new Qt3DRender::QBuffer( this ) )
 #endif
+  , mRenderingStyle( symbol->renderingStyle() )
 {
+  unsigned int byte_stride = mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering ? 28 : 16;
   mPositionAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
   mPositionAttribute->setBuffer( mVertexBuffer );
   mPositionAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
   mPositionAttribute->setVertexSize( 3 );
   mPositionAttribute->setName( Qt3DRender::QAttribute::defaultPositionAttributeName() );
   mPositionAttribute->setByteOffset( 0 );
-  mPositionAttribute->setByteStride( 16 );
+  mPositionAttribute->setByteStride( byte_stride );
 
-  mClassAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
-  mClassAttribute->setBuffer( mVertexBuffer );
-  mClassAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
-  mClassAttribute->setVertexSize( 1 );
-  mClassAttribute->setName( "cls" );
-  mClassAttribute->setByteOffset( 12 );
-  mClassAttribute->setByteStride( 16 );
+  mParameterAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
+  mParameterAttribute->setBuffer( mVertexBuffer );
+  mParameterAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
+  mParameterAttribute->setVertexSize( 1 );
+  mParameterAttribute->setName( "vertexParameter" );
+  mParameterAttribute->setByteOffset( 12 );
+  mParameterAttribute->setByteStride( byte_stride );
 
   addAttribute( mPositionAttribute );
-  addAttribute( mClassAttribute );
+  addAttribute( mParameterAttribute );
 
+  if ( mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering )
+  {
+    mColorAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
+    mColorAttribute->setBuffer( mVertexBuffer );
+    mColorAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
+    mColorAttribute->setVertexSize( 3 );
+    mColorAttribute->setName( QStringLiteral( "vertexColor" ) );
+    mColorAttribute->setByteOffset( 16 );
+    mColorAttribute->setByteStride( byte_stride );
+    addAttribute( mColorAttribute );
+  }
 
   makeVertexBuffer( data );
 }
 
 void QgsPointCloud3DGeometry::makeVertexBuffer( const QgsPointCloud3DSymbolHandler::PointData &data )
 {
   QByteArray vertexBufferData;
-  vertexBufferData.resize( data.positions.size() * 4 * sizeof( float ) );
+  if ( mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering )
```

Comment:
Here also -- you could calculate this by adding a method to QgsPointCloud3DSymbol for "attributeCount" or similar and keep the symbol class properties self contained.